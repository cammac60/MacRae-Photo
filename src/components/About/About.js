import React from 'react';
import { Link } from 'react-router-dom';

import './About.scss';

export const About = () => {
  return (
    <div className="About">
      <div className="about-back">
        <div className="blur-wrapper">
          <div className="about-wrapper">
            <div className="bio-wrapper">
              <img alt="Photo of Cameron MacRae" className="profile-img" src={require('../../images/IMG_5947.jpg')} />
              <div className="bio-text-wrapper">
                <p className="bio-text">
                I am a hobbyist photographer based in Denver, Colorado who specializes in landscape and cityscape. My passion for photography comes from my love of travelling and geography. Being able to document the world around me inspires me to see as much as possible and to step outside of my comfort zone. Some of my bucket list locations to shoot are Japan, San Marino, Morocco, New Zealand, Chile, and Nepal among many others.
                </p>
                <br/>
                <p className="bio-text">
                I generally donâ€™t shoot with a theme in mind but I enjoy distoring my photos and using unorthodox angles to create simple and somewhat minimalistic compositions. Although I typically shoot landscape and cityscape I also enjoy street, fine art, portraiture, and journalistic photography.
                </p>
                <br/>
                <p className="bio-text">
                Outside of photography I work as a software engineer and enjoy weight lifting, hiking, and hockey.
                </p>
              </div>
            </div>
            <div className="bio-footer">
              <h2 className="bio-name">Cameron MacRae</h2>
              <Link to="/" className="back-btn">Back</Link>
            </div>
          </div>
        </div>
      </div>
    </div>
  )
};

export default About;
