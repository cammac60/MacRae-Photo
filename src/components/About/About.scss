@import '../../variables.scss';

.about {
  height: 100vh;
  width: 100vw;

  &__background {
    background: {
      image: url('../../images/main/rmnp2.JPG');
      position: center;
      repeat: no-repeat;
      size: cover;
    }
    height: 100%;
    position: fixed;
    width: 100%;
  }

  &__wrapper {
    background: {
      color: rgba(0, 0, 0, 0.7);
      image: url('../../images/main/rmnp2.JPG');
      position: center;
      repeat: no-repeat;
      size: cover;
    }
    background-blend-mode: darken;
    border: 1px solid $main-color;
    height: 675px;
    margin: {
      right: auto;
      top: auto;
    }
    overflow: visible;
    padding: 2%;
    width: 95%;
  }

  &__bio-wrapper {
    display: flex;
    justify-content: space-between;
    margin: {
      left: auto;
      right: auto;
    }
    width: 95%;
  }

  &__profile-img {
    border: 1px solid $main-color;
    height: 500px;
    position: relative;
    top: -70px;
  }

  &__text-wrapper {
    display: block;
    font: {
      size: 1.4rem;
      weight: 300;
    }
    text-align: left;
    width: 55%;
  }

  &__text {
    text-indent: 20px;
    width: 100%;
  }

  &__footer {
    display: flex;
    justify-content: space-between;
    margin: {
      left: auto;
      right: auto;
      top: auto;
    }
    width: 95%;
  }

  &__name {
    font: {
      family: $large-font;
      size: 5rem;
      weight: normal;
    }
    margin-top: auto;
    max-height: 85px;
  }

  &__button {
    font: {
      size: 2rem;
      weight: 300;
    }
    margin-top: auto;
    text: {
      align: right;
      decoration: none;
    }
    &:hover {
      color: $secondary-accent-1;
    }
  }

}


@media (max-width: 1150px) {
  .about {

    &__text {
      font-size: 1.2rem;
    }

    &__profile-img {
      height: 450px;
    }

    &__name {
      font-size: 4rem;
    }

    &__footer {
      margin-top: 100px;
    }

  }
}


@media (max-width: 1025px) {
  .about {

    &__text {
      font-size: 1.1rem;
    }

    &__profile-img {
      height: 400px;
    }

  }
}

@media (max-width: 900px) {
  .about {

    &__text {
      font-size: 0.9rem;
    }

    &__profile-img {
      height: 350px;
    }

    &__footer {
      margin-top: 200px;
    }

  }
}

@media (max-width: 740px) {
  .about {

    &__text {
      font-size: 0.7rem;
    }
    
    &__profile-img {
      height: 300px;
    }

    &__footer {
      margin-top: 250px;
    }

    &__name {
      font-size: 3rem;
    }

  }
}
