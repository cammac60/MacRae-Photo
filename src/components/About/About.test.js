import React from 'react';
import { shallow } from 'enzyme';
import About from './About';

describe('About', () => {

  let wrapper, instance;

  beforeEach(() => {
    wrapper = shallow(<About />);
    instance = wrapper.instance();
  });

  it('Should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
