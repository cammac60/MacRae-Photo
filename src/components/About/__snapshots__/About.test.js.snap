// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Should match the snapshot 1`] = `ShallowWrapper {}`;
